docs/website/contents/for-developers/HardForkCombinatorRollbackMonotonicityProposal.md

```diff
@@ -0,0 +1,285 @@
+(In the non-simplified node, the prediction is always restricted to exactly 3k/f slots, despite the data in question actually advancing an entire epoch at a time.
+Thus regardless of where the first header is, any Chain Growth violation in the 3k/f slots after it would prevent switch to C when R=k.)
+
+That is a complete answer, but there are a few particulars worth emphasizing.
+
+- No other behavior of the simplified node will change due to a Chain Growth violation.
+  (This is actually false, but only because we use predictions in the leadership check.
+  One option to make this true is to instead use ticking in the leadership check.
+  Another option is to also characterize here how Chain Growth violations affect the predication-based leadership check, which is even milder than the above: it only manifests when there are zero blocks in some 3k/f slot window, not merely any number of blocks less than k.)
+- It's worth explicitly stating that there's never a risk of the node incorrectly judging the validity of a header.
+  At worst, the node won't be able to make any judgement of the header at all, since it will recognize that doing so correctly would require information that it cannot predict.
+- It's entirely possible the node's behavior wouldn't even reveal that there was a Chain Growth violation!
+  It's only guaranteed to fail if the correct Ouroboros Praos behavior were to require a k-deep roll back, which itself is an event may remain extremely rare despite whatever conditions caused the Chain Growth violation.
+  For example, suppose 90% of stake suddenly shut down their nodes, but somehow the remaining 10% was still well-connected and there remained a sufficiently dominant honest majority.
+  With just 10% of stake, it's almost certain that there would be less than k blocks in 3k/f slots, ie a Chain Growth violation.
+  However, it's also almost certain there would be hundreds of blocks in every window of 3k/f slots.
+  And, since the remaining nodes are still sufficiently well-connected, Praos's natural short forks would indeed remain short.
+  Thus R would remain much smaller than k, such that any node that needed to switch from D to C would still be able to validate R+1 headers after L on C despite not being able to predict more than 3k/f after the first header after L on C.
+
+## Back to Reality: the Cardano Node
+
+First, instead of merely pure Praos, consider a node running Cardano's hard fork combinator and its history of eras from Byron through Babbage.
+The answer to Question 1 actually does not change.
+One crucial reason is because Cardano sets the hard fork combinator safe zone of each era to that era's stability window.
+If the safe zone were instead less than one stability window, then L does not necessarily contain enough information to validate the R+1th header after L on C, _even without_ a Chain Growth violation.
```

Comment:
On the Byron->Shelley transition, it is also relevant that forecasting from a Byron blocks within 2k slots of the era transition into Shelley has a range much *larger* than one Byron stability window, otherwise this property would be violated.

Reply:
To double check: is it true that to judge the validity of headers we need to be within the forecast range?

Reply:
Also, according to my notes, it must always be the case that

```
forecast range <= safe zone
```

This makes sense since we cannot forecast if we can't guarantee a transition won't happen. 

What I still don't get (but it's outside the scope of the PR, but might be related to Alex's comment) is what happens to the forecast range around an era transition.